sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -343,8 +343,8 @@ object QueryExecution {
       PlanDynamicPruningFilters,
       PlanSubqueries,
       RemoveRedundantProjects,
-      RemoveRedundantSorts,
       EnsureRequirements,
+      RemoveRedundantSorts,
```

Comment:
Could you leave some comments here about why we need to put this rule after `EnsureRequirements`?

Reply:
+1 for @maropu 's comment.